plugin/db/util/driverutil.go

```diff
@@ -209,9 +209,6 @@ func BeginMigration(ctx context.Context, executor MigrationExecutor, m *db.Migra
 		case db.Pending:
 			return -1, common.Errorf(common.MigrationPending,
 				fmt.Errorf("database %q version %s migration is already in progress", m.Database, m.Version))
-		case db.Failed:
```

Comment:
The reason we do this is that when failure happens, the DB may be in an inconsistent state, and it requires someone to manually check and proceed with caution.